src/collision/Distance.ts

```diff
@@ -707,3 +716,170 @@ Distance.Input = DistanceInput;
 Distance.Output = DistanceOutput;
 Distance.Proxy = DistanceProxy;
 Distance.Cache = SimplexCache;
+
+/**
+ * Input parameters for ShapeCast
+ */
+export class ShapeCastInput {
+  proxyA: DistanceProxy = new DistanceProxy();
+  proxyB: DistanceProxy = new DistanceProxy();
+  transformA: Transform | null = null;
+  transformB: Transform | null = null;
+  translationB: Vec2 = Vec2.zero();
+}
+
+/**
+ * Output results for b2ShapeCast
+ */
+export class ShapeCastOutput {
+  point: Vec2 = Vec2.zero();
+  normal: Vec2 = Vec2.zero();
+  lambda: number;
+  iterations: number;
+}
+
+/**
+ * Perform a linear shape cast of shape B moving and shape A fixed. Determines
+ * the hit point, normal, and translation fraction.
+ * @returns true if hit, false if there is no hit or an initial overlap
+ */
+//
+// GJK-raycast
+// Algorithm by Gino van den Bergen.
+// "Smooth Mesh Contacts with GJK" in Game Physics Pearls. 2010
+export const ShapeCast = function(output: ShapeCastOutput, input: ShapeCastInput): boolean {
```

Comment:
could we put this in its own file?

Reply:
We would only have to export Simplex from Distance.ts. Would that be ok? Or should I move Simplex into its own file?

Reply:
Nevermind actually, this is good, thanks